fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/TerribleIndexMaintainer.java

```diff
@@ -94,6 +94,12 @@ public <M extends Message> CompletableFuture<Void> update(@Nullable FDBIndexable
         }
     }
 
+    @Nonnull
+    @Override
+    public <M extends Message> CompletableFuture<Void> updateWhileWriteOnly(@Nullable final FDBIndexableRecord<M> oldRecord, @Nullable final FDBIndexableRecord<M> newRecord) {
+        return AsyncUtil.DONE;
```

Comment:
Should this just call `update`? or return a failed exception

Reply:
I had this return `DONE` rather than throw an error because this index maintainer always returns `false` to `isIdempotent`, so this is behavior preserving. I didn't try throwing an error, but IIRC, this maintainer is used in some somewhat delicate tests, so it could cause some tests to break, I think